### PR TITLE
SITES-508 send confirmation emails

### PR DESCRIPTION
**Pull request by freshtonic**
Production & staging require the following environment variables:

(NOTE: still waiting on production AWS env vars to be configured).

- `AWS_ACCESS_KEY`
- `AWS_SECRET_KEY`
- `FROM_EMAIL`
- `APP_DOMAIN`

### CODE REVIEW

**Comment by freshtonic** [comment]
The `letter_opener` gem opens confirmation emails in your browser in the development environment.

It's worth noting that the confirmation email text is simply the Devise default right now.

It looks like this:

image

File: config/environments/development.rb [hunk 1]
```diff
@@ -33,6 +33,8 @@
 
   config.action_mailer.default_url_options = { host: 'localhost', port: 3000 }
 
+  config.action_mailer.delivery_method = :letter_opener
```
Comment (tessereth) [inline config/environments/development.rb]: When I was doing request for membership emails, I set up mailtrap. See `config/initializers/mailer.rb` and the readme. How does this interact with mailtrap? Would this replace it?

Comment (freshtonic) [inline config/environments/development.rb]: @tessereth yes, I noticed after the fact.

So letteropener runs entirely locally and when an email is sent in in the development env it opens a browser tab containing the email.

I'm open to using mailtrap but I would rather not depend on a 3rd party service unless there's other compelling reasons to do so.

File: config/initializers/amazon_ses.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+
+if Rails.env.production?
+  ActionMailer::Base.add_delivery_method(:ses, AWS::SES::Base,
+    access_key_id:      ENV['AMAZON_ACCESS_KEY'],
+    secret_access_key:  ENV['AMAZON_SECRET_KEY']
```
Comment (ltankey) [inline config/initializers/amazon_ses.rb]: Do we bother having defaults if the environment doesn't provide these details?

File: db/schema.rb [hunk 3]
```diff
@@ -10,11 +10,19 @@
...
   enable_extension "plpgsql"
 
+  create_table "content_blocks", force: :cascade do |t|
+    t.integer  "node_id"
+    t.text     "body"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["node_id"], name: "index_content_blocks_on_node_id", using: :btree
+  end
+
```
Comment (ltankey) [inline db/schema.rb]: Please kill this with fire.

Comment (freshtonic) [inline db/schema.rb]: OH FFS it's like an unkillable whack-a-mole @ltankey

**Comment by ltankey** [comment]
LGTM

File: config/environments/production.rb [hunk 4]
```diff
@@ -64,6 +64,13 @@
   # Set this to true and configure the email server for immediate delivery to raise delivery errors.
   # config.action_mailer.raise_delivery_errors = false
 
+  config.action_mailer.delivery_method = :ses
+
+  config.action_mailer.default_url_options = {
+    protocol: 'https',
+    host: 'gov-au-beta.apps.platform.digital.gov.au'
```
Comment (tessereth) [inline config/environments/production.rb]: This should come from environment as this is not 12 factor

Comment (freshtonic) [inline config/environments/production.rb]: @tessereth yes it should - do we already have an env var for that?

Comment (ltankey) [inline config/environments/production.rb]: @freshtonic you can access them via `VCAP_APPLICATION` environment variable which should contain JSON we need to parse (https://docs.cloudfoundry.org/devguide/deploy-apps/environment-variable.html#app-system-env)

We'd need to access `application-uris` and regex out any coloured URLs (-blue and -green variants). The remaining uri would be the one we want to keep. 

**note** during our deploy, we're likely to remove the route to our app (which would leave _only_ the coloured variant URI). As such, we'd need to use regex to remove the colour from the URI we have access to.

File: config/initializers/amazon_ses.rb (hunk 2 above)
Comment (freshtonic) [inline config/initializers/amazon_ses.rb]: What should happen is that the app refuses to start and produces a helpful error message detailing all missing configuration.

But nothing else in the app does that or provides defaults.

We should revisit later and flag as tech debt.

File: db/migrate/20160801053501_add_confirmed_at_to_users.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+class AddConfirmedAtToUsers < ActiveRecord::Migration[5.0]
```
Comment (maxious) [inline db/migrate/20160801053501_add_confirmed_at_to_users.rb]: What happens if an account existed before these columns were created? Will they be able to log in?

Comment (freshtonic) [inline db/migrate/20160801053501_add_confirmed_at_to_users.rb]: Amazing catch - no they won't!

Aren't code reviews _awesome_!

Comment (freshtonic) [inline db/migrate/20160801053501_add_confirmed_at_to_users.rb]: @maxious I can fix it in the DB but it does mean that we will *never* send confirmation emails for existing users. Are we OK with that?

Comment (maxious) [inline db/migrate/20160801053501_add_confirmed_at_to_users.rb]: I think so. Maybe include a date created filter before today or something?

**Comment by tessereth** [comment]
Update administrate with the new user columns?

**Comment by tessereth** [comment]
LGTM pending testing in staging

**Pull request merged**